core/src/main/java/org/apache/iceberg/hadoop/HadoopStreams.java

```diff
@@ -185,8 +185,20 @@ public void flush() throws IOException {
 
     @Override
     public void close() throws IOException {
-      stream.close();
-      this.closed = true;
+      try {
+        stream.close();
+        // {@link org.apache.hadoop.fs.s3a.S3ABlockOutputStream#close()} calls {@link
+        // org.apache.hadoop.fs.s3a.S3ABlockOutputStream#putObject()}
+        // which doesn't throw an exception when interrupted. Need to handle interrupts here to make
+        // sure file is uploaded on close.
+        if (Thread.interrupted()
+            && "S3ABlockOutputStream"
```

Comment:
I can remove this explicit class name check and just check for an interrupt.

This will make it easier to test this with just unit test

Reply:
Could we create a class with a same name in a different package?

Reply:
@pvary
you mean we should include the full class name with package in the if check to cover for other classes with same name?

Reply:
+1 on the fully qualified name. or maybe just use `instanceOf`.

Reply:
I didn't want to add hadoop-aws module to use `instanceOf`. Let me update this to fully qualified name.

Reply:
updated to fully qualified name instead. cannot use `instanceOf` as `org.apache.hadoop.fs.s3a.S3ABlockOutputStream` is package protected.